src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkAppEngineFlexibleStaging.java

```diff
@@ -48,15 +49,25 @@ public void stageFlexible(StageFlexibleConfiguration config) throws AppEngineExc
 
     if (!config.getStagingDirectory().exists()) {
       throw new AppEngineException("Staging directory does not exist. Location: "
-          + config.getStagingDirectory().toPath().toString());
+          + config.getStagingDirectory().toPath());
     }
     if (!config.getStagingDirectory().isDirectory()) {
       throw new AppEngineException("Staging location is not a directory. Location: "
-          + config.getStagingDirectory().toPath().toString());
+          + config.getStagingDirectory().toPath());
     }
 
     try {
 
+      // Copy docker context to staging
+      if (config.getDockerDirectory() != null && config.getDockerDirectory().exists()) {
+        if (!Files.isRegularFile(config.getDockerDirectory().toPath().resolve("dockerfile"))) {
```

Comment:
case sensitivity is going to be an issue here too.  The convention is that the file is called 'Dockerfile'